src/backend/distributed/connection/connection_management.c

```diff
@@ -437,184 +457,378 @@ ShutdownConnection(MultiConnection *connection)
 }
 
 
-/*
- * FinishConnectionListEstablishment is a wrapper around FinishConnectionEstablishment.
- * The function iterates over the multiConnectionList and finishes the connection
- * establishment for each multi connection.
- */
-void
-FinishConnectionListEstablishment(List *multiConnectionList)
+static enum EventSetResult
+MultiConnectionStatePoll(MultiConnectionState *connectionState)
 {
-	ListCell *multiConnectionCell = NULL;
+	MultiConnection *connection = connectionState->connection;
+	ConnStatusType status = PQstatus(connection->pgConn);
+	PostgresPollingStatusType oldPollmode = connectionState->pollmode;
 
-	foreach(multiConnectionCell, multiConnectionList)
+	Assert(!connectionState->ready);
+
+	if (status == CONNECTION_OK)
 	{
-		MultiConnection *multiConnection = (MultiConnection *) lfirst(
-			multiConnectionCell);
+		connectionState->ready = true;
+		return Rebuild;
+	}
+	else if (status == CONNECTION_BAD)
+	{
+		/* FIXME: retries? */
+		connectionState->ready = true;
```

Comment:
Was this meant to be set to `false`? If not, I'm a bit confused why we're setting this to `true`.

Reply:
I think because we're no longer waiting for the connection. Perhaps this should be an enum instead of a bool to avoid confusion.